File: project/MagistrProject/Event.cpp
#include "classes.h"
using namespace Input;
using namespace Input::Event;

//Список событий

CEvent::CEvent()
	{
		sourceTable = "Event";
	};


//Возврщает новый id для вставки записи в БД
int CEvent::NewID()
	{
		int result;
		string queryText;
		queryText="SELECT MAX(idEvent) FROM "+sourceTable;
		DoQuery(queryText);

		result=25;// для примера
		return ++result; 
	};


//Добавление события
void CEvent::Add(string name)
	{
		string queryText;
		queryText="INSERT INTO "+sourceTable;
		queryText.append("(idEvent, name) ");
		queryText.append("VALUES (");
		queryText.append(IntToStr(NewID()));
		queryText.append(",");
		queryText.append(name);
		queryText.append(")");
		DoQuery(queryText);
	};

//Удаление необходимого события по id
void CEvent::Del(int id)
	{
		string queryText;
		queryText="DELETE FROM "+sourceTable;
		queryText.append(" WHERE idEvent = ");
		queryText.append(IntToStr(id));
		DoQuery(queryText);
	};

//Изменение записи о событии
void CEvent::Update(int id, string newName)
	{
		string queryText;
		queryText="UPDATE "+sourceTable;
		queryText.append(" SET name = ");
		queryText.append(newName);
		queryText.append(" WHERE idEvent = ");
		queryText.append(IntToStr(id));
		DoQuery(queryText);
	};

//Вывод списка событий
void CEvent::Show()
	{
		string queryText;
		queryText="SELECT * FROM "+sourceTable;
		DoQuery(queryText);
	};

File: project/CSharpMagistrProject/CSharpMagistrProject/Input/Event/Event.cs
using System.Windows.Forms;
using CSharpMagistrProject.DB;

namespace CSharpMagistrProject.Input.Event
{
    //Список событий
    class Event
    {
        private string sourceTable;
        private DataBase dataBase;

        public Event(DataBase sourceDataBase, string sourceTable)
        {
            dataBase = sourceDataBase;
            dataBase.Connect();
            this.sourceTable = sourceTable;
        }

		//Возврщает новый id для вставки записи в БД
        private int NewID()
        {
            if (dataBase.isConnected==true)
            {
                string queryText;
                queryText = "SELECT MAX(idEvent) FROM " + sourceTable;
                int result = dataBase.DoScalarQuery(queryText);
                if (result!=-1)
                {
                    return ++result; 
                }
            }
            return -1;
        }

		//Добавление события
        public void Add(string nameEvent)
        {
            string queryText;
            queryText = "INSERT INTO " + sourceTable + "(idEvent, name) ";
            queryText += "VALUES (" + NewID().ToString() + "," +@"""" +nameEvent + @""")";
            dataBase.DoQuery(queryText);
        }

        //Удаление необходимого события по id
        public void Del(int id)
        {
            string queryText;
            queryText = "DELETE FROM " + sourceTable;
            queryText += " WHERE idEvent = "+ id.ToString();
            dataBase.DoQuery(queryText);
        }

		//Изменение записи о событии
        public void Update(int id, string newName)
        {
            string queryText;
            queryText = "UPDATE " + sourceTable;
            queryText += @" SET name = """ + newName+ @"""";
            queryText += " WHERE idEvent = " + id.ToString();
            dataBase.DoQuery(queryText);
        }

		//Вывод списка событий
        public void Show(DataGridView receiverGridView)
        {
            string queryText;
            queryText = "SELECT * FROM " + sourceTable;
            dataBase.DoQuery(queryText,receiverGridView);
        }
    }
}
